lib/context_adapter_server.js

```diff
@@ -0,0 +1,427 @@
+  var hapi = require('hapi');
+  var boom = require('boom');
+
+  var server;
+
+  var attendedRequests = 0;
+
+  /**
+   * Checks that every request sent to the Context Adapter includes the required headers
+   * @param {object} value Headers object
+   * @param {object} options Hapi server header validation configuration object
+   * @param {function} next Hapi server header validation continuation function
+   */
+  function validateHeaders(value, options, next) {
+    var error, message;
+
+    var meta = {
+      unicaCorrelator: value[caConfig.UNICA_CORRELATOR_HEADER] || caHelper.getUnicaCorrelator(),
+      transactionId: caHelper.getTransactionId(),
+      operationType: caHelper.getOperationType()
+    };
+
+    attendedRequests++;
+
+    if (!value['fiware-service']) {
+      message = 'error=child "fiware-service" fails because [fiware-service is required]';
+      caLogger.warn(
+        message,
+        meta
+      );
+      error = boom.badRequest(message);
+      error.output.payload.validation = {source: 'headers', keys: ['fiware-service']};
+      next(error);
+    } else if (!value['fiware-servicepath']) {
+      message = 'child "fiware-servicepath" fails because [fiware-servicepath is required]';
+      caLogger.warn(
+        message,
+        meta
+      );
+      error = boom.badRequest(message);
+      error.output.payload.validation = {source: 'headers', keys: ['fiware-servicepath']};
+      next(error);
+    }
+    next();
+  }
+
+  /**
+   * Starts the server asynchronously
+   * @param {string} host The Context Adapter server host
+   * @param {string} port The Context Adapter server port
+   * @param {Function} callback Callback function to notify the result
+   *  of the operation
+   */
+  function start(host, port, callback) {
+    server = new hapi.Server();
+
+    server.on('log', function(event, tags) {
+      if (tags.load) {
+        caLogger.warn('event=' + JSON.stringify(event), {
+          operationType: caConfig.OPERATION_TYPE.SERVER_LOG
+        });
+      }
+    });
+
+    server.on('request-internal', function(request, event, tags) {
+      if (tags.error) {
+        if (tags.auth || tags.handler || tags.state || tags.payload || tags.validation) {
+          caLogger.warn(
+            request.method.toUpperCase() + ' ' + request.url.path +
+            ', event=' + JSON.stringify(event), {
+              operationType: caConfig.OPERATION_TYPE.SERVER_LOG
+            }
+          );
+        } else {
+          caLogger.error(
+            request.method.toUpperCase() + ' ' + request.url.path +
+            ', event=' + JSON.stringify(event), {
+              operationType: caConfig.OPERATION_TYPE.SERVER_LOG
+            }
+          );
+        }
+      }
+    });
+
+    server.connection({
+      host: host,
+      port: port
+    });
+
+    server.route([
+      {
+        method: 'GET',
+        path: '/version',
+        handler: function(request, reply) {
```

Comment:
You can sepparate this handlers from the mapping itself, and the code will be much more clear. Give a name to each handler and put it appart in the file (or even in different files if this file ends up being too big.